perception/object_recognition/tracking/multi_object_tracker/src/multi_object_tracker_core.cpp

```diff
@@ -0,0 +1,200 @@
+  world_frame_id_ = declare_parameter<std::string>("world_frame_id", std::string("world"));
+
+  rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_ROS_TIME);
+  tf_buffer_ = std::make_shared<tf2_ros::Buffer>(clock);
+  tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+
+  auto cti = std::make_shared<tf2_ros::CreateTimerROS>(this->get_node_base_interface(), this->get_node_timers_interface());
+  tf_buffer_->setCreateTimerInterface(cti);
+
+  // Create ROS time based timer
+  auto timer_callback = std::bind(&MultiObjectTracker::publishTimerCallback, this);
+  auto period = std::chrono::duration_cast<std::chrono::nanoseconds>(
+    std::chrono::duration<double>(publish_rate));
+
+  publish_timer_ = std::make_shared<rclcpp::GenericTimer<decltype(timer_callback)>>(
+    this->get_clock(), period, std::move(timer_callback),
+    this->get_node_base_interface()->get_context());
+  this->get_node_timers_interface()->add_timer(publish_timer_, nullptr);
+}
+
+void MultiObjectTracker::measurementCallback(
+  const autoware_perception_msgs::msg::DynamicObjectWithFeatureArray::ConstSharedPtr
+    input_objects_msg)
+{
+  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray input_transformed_objects =
+    *input_objects_msg;
+
+  /* transform to world coordinate */
+  if (input_objects_msg->header.frame_id != world_frame_id_) {
+    tf2::Transform tf_world2objects_world;
+    tf2::Transform tf_world2objects;
+    tf2::Transform tf_objects_world2objects;
+    try {
+      geometry_msgs::msg::TransformStamped ros_world2objects_world;
+      auto future = tf_buffer_->waitForTransform(
+        world_frame_id_, input_transformed_objects.header.frame_id,
+        tf2::TimePoint(
+          std::chrono::seconds(input_transformed_objects.header.stamp.sec) +
+          std::chrono::nanoseconds(input_transformed_objects.header.stamp.nanosec)),
+        tf2::durationFromSec(0.5),
+        [&ros_world2objects_world](
+          const std::shared_future<geometry_msgs::msg::TransformStamped> & future) {
+          ros_world2objects_world = future.get();
+        });
+      tf2::fromMsg(ros_world2objects_world.transform, tf_world2objects_world);
+    } catch (tf2::TransformException & ex) {
+      RCLCPP_WARN(this->get_logger(), "%s", ex.what());
+      return;
+    }
+    for (size_t i = 0; i < input_transformed_objects.feature_objects.size(); ++i) {
+      tf2::fromMsg(
+        input_transformed_objects.feature_objects.at(i).object.state.pose_covariance.pose,
+        tf_objects_world2objects);
+      tf_world2objects = tf_world2objects_world * tf_objects_world2objects;
+      tf2::toMsg(
+        tf_world2objects,
+        input_transformed_objects.feature_objects.at(i).object.state.pose_covariance.pose);
+    }
+  }
+
+  /* tracker prediction */
+  rclcpp::Time measurement_time = input_objects_msg->header.stamp;
+  for (auto itr = list_tracker_.begin(); itr != list_tracker_.end(); ++itr) {
```

Comment:
Really surprised that this isn't a range-based for. I find range-based for much easier to read. I know it's a style issue though, so feel free to leave it like it is.

Reply:
I totally agree with you, I didn't find these all that easy to read either. As far as I'm concerned there are plenty of other places and packages which could benefit from these stylistic changes. Maybe we could create an issue for this and append any other stylistic changes to work on there? If you're ok with this I can create the issue

Reply:
Sure, I'm ok with an issue

Reply:
Here is the issue: https://github.com/tier4/Pilot.Auto/issues/61